pipelines/nyu/adapters/course_catalog/src/process_data.rb

```diff
@@ -1,60 +1,101 @@
 require 'pry'
+require 'uri'
 
 # List of courses for a subject
 def extract_listings html
-	elements = html.children
-	elements[0] = Hash.new { |h,k| h[k] = [] }
-  binding.pry
-	# elements.inject do |first, second|
-	# 	nil
-	# end
-	nil
+  elements = html.children[1].child.children[1..]
+  current = nil
+  elements.map do |item|
+    if item.name == 'div'
+      current = []
+      parse_course_header item,current
+    else
+      if item.name == 'a'
+        current << (parse_section item)
+      end
+      nil
+    end
+  end.select { |item| item.is_a? Hash }
 end
 
-def parse_header div_tag
-
+def parse_course_header div_tag,sections
+  shortname_raw, longname = div_tag.text.split ' - '
+  shortname = shortname_raw.split(' ')[1]
+  { # Yacs format
+    "shortname" => shortname,
+    "longname" => longname,
+    "min_credits" => nil,
+    "max_credits" => nil,
+    "description" => nil,
+    "sections" => sections
+  }
 end
 
 def parse_section a_tag
+  course_url = a_tag.attributes['href'].value
+  crn = URI(course_url).path.chomp('/').split('/').last
+
+  data_div = a_tag.children.find { |element| element.name == 'div' }
```

Comment:
Try to use selectors or xpaths rather than `find`

Reply:
replaced with `css('div.section-content')`